py/miniconf-mqtt/miniconf/__main__.py

```diff
@@ -67,36 +69,38 @@ def main():
     # If a discovery was requested, try to find a device.
 
     async def run():
-        client = Client(client_id="")
-        await client.connect(args.broker)
-
-        if args.discover:
-            devices = await discover(client, args.prefix)
-            if len(devices) != 1:
-                raise MiniconfException(
-                    f"No unique Miniconf device (found `{devices}`). Please specify a `--prefix`"
-                )
-            prefix = devices.pop()
-            logging.info("Found device prefix: %s", prefix)
-        else:
-            prefix = args.prefix
-
-        interface = await Miniconf.create(client, prefix)
-
-        for arg in args.paths:
-            try:
-                path, value = arg.split("=", 1)
-            except ValueError:
-                value = await interface.get(arg)
-                print(f"{arg} = {value}")
+        async with Client(args.broker,
+                          protocol=MQTTv5,
+                          logger=logging.getLogger(__name__)) as client:
+            if args.discover:
+                devices = await discover(client, args.prefix)
+                if len(devices) != 1:
+                    raise MiniconfException(
+                        f"No unique Miniconf device (found `{devices}`). "
+                        "Please specify a `--prefix`"
+                    )
+                prefix = devices.pop()
+                logging.info("Found device prefix: %s", prefix)
             else:
-                await interface.set(path, json.loads(value), args.retain)
-                print(f"Set {path}: OK")
-
-        if args.list:
-            for path in await interface.list_paths():
-                value = await interface.get(path)
-                print(f"{path} = {value}")
+                prefix = args.prefix
+
+            interface = Miniconf(client, prefix)
+
+            for arg in args.paths:
+                try:
+                    path, value = arg.split("=", 1)
+                except ValueError:
+                    value = await interface.get(arg)
+                    print(f"{arg} = {value}")
+                else:
+                    print(f'Setting {arg} to {value}')
```

Comment:
```suggestion
```

Or merge with the print in L98.
Or `log::info()`

Reply:
I'd rather not mix in `log` here because there's a lot of MQTT traffic that gets logged if you go down to `info`